src/test/kotlin/racingCar/entity/CarImplTest.kt

```diff
@@ -0,0 +1,49 @@
+package racingCar.entity
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+
+internal class CarImplTest {
+
+    @Test
+    fun `자동차가 이동하면 distance가 1 증가한다`() {
+        // given
+        val car = CarImpl()
```

Comment:
```suggestion
        val car: Car = CarImpl()
```
인터페이스를 사용하신다면, 인터페이스 타입으로 구현부를 감출 수 있어요. (이하 관련 내용 동일)

Reply:
`@BeforeEach` 어노테이션을 활용하면, 테스트 코드가 실행 되기 전에 매번 호출해서 Car와 같은 클래스들을 매번 초기화하도록 할 수 있어요.

Reply:
꼼꼼하게 봐주셔서 정말 감사드려요! 혹시 여기에서 구현부를 숨길 수 있다고 말씀해주신 게 어떤 뜻인지 여쭤볼 수 있을까요?

Reply:
@jayleeinflab 
위 코드를 예로 들자면, car는 `Car` 가 아닌 `CarImpl` 타입이 됩니다.
car가 `CarImpl` 타입이므로, `car.move()`가 어떻게 구현이 되어있는지 볼 수 있다는 것을 의미합니다.
하지만 타입을 `Car`로 하게 되면, `car.move()`가 어떻게 구현이 되어있는지 알 수 없습니다.
어떻게 구현되어있는지 알 수 없기 때문에, 구현체의 코드가 변경되어도 영향을 받지 않습니다.

Reply:
그렇군요!! 모르고 있던 부분인데 알려주셔서 감사합니다!! 🙌 👍

Reply:
말씀해주신대로 @beforeEach를 사용해보려고 하다보니, 이렇게 공통되는 부분을 빼서 설정하다보면 클래스에 있는 하나의 변수에 여러 테스트들이 의존성이 생길 것 같더라구요. 결과적으로 하나의 클래스에 이 테스트 픽스쳐를 사용하는 모든 테스트들의 의존성이 생기는 것 같아요. 또 만약에 여기있는 함수를 다른 곳으로 옮긴다면 관련 픽스쳐를 다시 적어야 할 것 같다는 생각도 듭니다.

저는 다음과 같이 공통적으로 사용하는 car 와 distace를 추출했습니다
새로 값을 assign 해야한다고 생각해서 var를 사용했습니다.
현석님은 어떻게 생각하시는 지 궁금합니다. 🙌 @BeokBeok  

```
internal class CarImplTest {
    lateinit var car: Car
    var expectedDistance: Int = 0

    @BeforeEach
    fun createCar(){
        car = CarImpl("A")
        expectedDistance = car.distance
    }

    @Test
    fun `자동차가 이동하면 distance가 1 증가한다`() {
        // given
        val moves = 6

        // when
        val testCar = car.move(moves)

        // then
        Assertions.assertThat(testCar.distance).isEqualTo(expectedDistance + 1)
    }

    @Test
    fun `move()에 4 이상의 숫자가 들어가면 자동차가 1 이동한다`() {
        // given
        val randomNumber = 6

        // when
        val testCar = car.move(randomNumber)

        // then
        Assertions.assertThat(testCar.distance).isEqualTo(expectedDistance + 1)
    }

   ...

```

Reply:
본 미션을 리뷰할 때에는 아래와 같은 코드가 테스트 코드마다 존재했습니다.
`val car = CarImpl()`

위와 같이 공통적으로 초기화해서 사용하는 경우에는 `@BeforeEach` 를 활용하시면 중복 코드를 최소화할 수 있어서 코맨트를 작성했습니다. 😅 
만약 매번 다른 값으로 초기화해야한다면 `@BeforeEach`를 사용하지 않아도 돼요.